lib/ansible/modules/files/file.py

```diff
@@ -370,6 +370,8 @@ def execute_touch(path, follow):
             raise
 
     # Unfortunately, touch always changes the file because it updates file's timestamp
+    if module.check_mode:
+        return {'dest': path, 'changed': True}
```

Comment:
@abadger Should I return a diff here like `{'before': {'path': None}, 'after': {'path': path},}`?

Reply:
@bcoca Do you mean this?
```
diff --git a/lib/ansible/modules/files/file.py b/lib/ansible/modules/files/file.py
index 712dbbc0fa..38fcca0153 100644
--- a/lib/ansible/modules/files/file.py
+++ b/lib/ansible/modules/files/file.py
@@ -369,10 +369,9 @@ def execute_touch(path, follow):
                     os.remove(b_path)
             raise

-    # Unfortunately, touch always changes the file because it updates file's timestamp
-    if module.check_mode:
-        return {'dest': path, 'changed': True}
-    return {'dest': path, 'changed': True, 'diff': diff}
+        # Unfortunately, touch always changes the file because it updates file's timestamp
+        return {'dest': path, 'changed': True, 'diff': diff}
+    return {'dest': path, 'changed': True}
```

Reply:
not really, I meant having a single point of return with 'all the keys' so you dont have a bunch of variations, also i would only return diff if module is in diff mode

Reply:
@bcoca better now?